woocommerce/woocommerce-android
# [Coupons] Product selector list

This PR is the first part of a product selector dialog (it's already too big, sorry about that), which is used in a conditional application of coupons to specific products.

Right now, only a list of products is displayed, with the products that are included in the coupon preselected. You can change the selection and the button title is updated, but it doesn't do anything yet. The search and filtering has also not been implemented yet.

**To test:**
1. Go to coupons, tap on one and go to the edit screen
2. Notice there are 2 new buttons, under the `Apply coupon to`: `All products ` and `Select (or Edit) products`
3. Currently, both just open the product selection
4. Verify the product list is loaded properly

Note the only regular products (not variations) are currently supported for selection.

![image](https://user-images.githubusercontent.com/1522856/169794772-c4a9e4a1-423f-4f74-af51-aa21143c170d.png)
![image](https://user-images.githubusercontent.com/1522856/169795520-6c7f02f4-012c-42c7-bffb-e49fca2136f1.png)
